libtransmission/rpcimpl.cc

```diff
@@ -2332,141 +2054,634 @@ void add_strings_from_var(std::set<std::string_view>& strings, tr_variant const&
+        {
+            if (auto const val = src.value_if<int64_t>())
+            {
+                tr_sessionSetAntiBruteForceThreshold(&tgt, static_cast<int>(*val));
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_blocklist_enabled,
+        [](tr_session const& src) -> tr_variant { return src.blocklist_enabled(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tgt.set_blocklist_enabled(*val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_blocklist_size,
+        [](tr_session const& src) -> tr_variant { return tr_blocklistGetRuleCount(&src); },
+        nullptr);
+
+    map.try_emplace(
+        TR_KEY_blocklist_url,
+        [](tr_session const& src) -> tr_variant { return src.blocklistUrl(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<std::string_view>())
+            {
+                tgt.setBlocklistUrl(*val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_cache_size_mib,
+        [](tr_session const& src) -> tr_variant { return tr_sessionGetCacheLimit_MB(&src); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<int64_t>())
+            {
+                tr_sessionSetCacheLimit_MB(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(TR_KEY_config_dir, [](tr_session const& src) -> tr_variant { return src.configDir(); }, nullptr);
+
+    map.try_emplace(
+        TR_KEY_default_trackers,
+        [](tr_session const& src) -> tr_variant { return src.defaultTrackersStr(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<std::string_view>())
+            {
+                tgt.setDefaultTrackers(*val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_dht_enabled,
+        [](tr_session const& src) -> tr_variant { return src.allowsDHT(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& /*err*/)
+        {
+            if (auto const val = src.value_if<bool>())
+            {
+                tr_sessionSetDHTEnabled(&tgt, *val);
+            }
+        });
+
+    map.try_emplace(
+        TR_KEY_download_dir,
+        [](tr_session const& src) -> tr_variant { return src.downloadDir(); },
+        [](tr_session& tgt, tr_variant const& src, ErrorInfo& err)
+        {
+            if (auto const download_dir = src.value_if<std::string_view>())
+            {
+                if (tr_sys_path_is_relative(*download_dir))
+                {
+                    err = { JsonRpc::Error::PATH_NOT_ABSOLUTE, "download directory path is not absolute"s };
+                }
+                else if (!std::empty(*download_dir))
```

Comment:
This check seems to be new. If we're going to keep it, I think we should return an error for it instead of silently ignoring the parameter.